src/forms/inputs/radio-group.js

```diff
@@ -137,6 +147,7 @@ function RadioGroup(props) {
     <LabeledField
       className={className}
       labelComponent={RadioGroupLegend}
+      as="fieldset"
       {...props}
     >
       {optionObjects.map((option, i) => {
```

Comment:
Might as well make the same changes made to [Checkbox Group](https://github.com/LaunchPadLab/lp-components/pull/595/files/036cf6d2d2be75b107bfa164dea4bb8a8c5fe274#r1334902078) here as well:
- Removing the lint disabling
- Pulling up `key` out of the props spread
- Setting `key` to `option.value` instead of index